### ISSUE TITLE
[8.0] [adhoc_modules] Breaks openerp --test-enable

### ISSUE THREAD

**Issue by ivantodorovich** [ingadhoc/odoo-support#25]
I took me a while to figure out the cause. Now I'm sure it's because of `adhoc_modules`.

I'm working on a dev database that's a copy of my production database, where this module is installed. While trying to run tests with `openerp-server --test-enable` I've encountered this error:

```
2016-11-05 14:37:54,323 1 CRITICAL binderplus openerp.service.server: Failed to initialize database `binderplus`.
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/openerp/service/server.py", line 936, in preload_registries
    registry = RegistryManager.new(dbname, update_module=update_module)
  File "/usr/lib/python2.7/dist-packages/openerp/modules/registry.py", line 370, in new
    openerp.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/usr/lib/python2.7/dist-packages/openerp/modules/loading.py", line 326, in load_modules
    modobj.button_upgrade(cr, SUPERUSER_ID, ids)
  File "/usr/lib/python2.7/dist-packages/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/openerp/addons/base/module/module.py", line 559, in button_upgrade
    if dep.state == 'unknown':
  File "/usr/lib/python2.7/dist-packages/openerp/fields.py", line 835, in __get__
    self.determine_value(record)
  File "/usr/lib/python2.7/dist-packages/openerp/fields.py", line 937, in determine_value
    self.compute_value(recs)
  File "/usr/lib/python2.7/dist-packages/openerp/fields.py", line 893, in compute_value
    self._compute_value(records)
  File "/usr/lib/python2.7/dist-packages/openerp/fields.py", line 885, in _compute_value
    self.compute(records)
  File "/usr/lib/python2.7/dist-packages/openerp/api.py", line 266, in wrapper
    return new_api(self, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/openerp/api.py", line 443, in new_api
    result = [method(rec, *args, **kwargs) for rec in self]
  File "/usr/lib/python2.7/dist-packages/openerp/addons/base/module/module.py", line 822, in _compute_state
    self.state = self.depend_id.state or 'unknown'
  File "/usr/lib/python2.7/dist-packages/openerp/fields.py", line 851, in __set__
    value = self.convert_to_cache(value, record)
  File "/usr/lib/python2.7/dist-packages/openerp/fields.py", line 1429, in convert_to_cache
    raise ValueError("Wrong value for %s: %r" % (self, value))
ValueError: Wrong value for ir.module.module.dependency.state: u'ignored'
```

As you can see, the test routine doesn't recognize the `module.dependency.state` introduced by `adhoc_modules`, so it throws an error.

My first thought was to just uninstall this module on my dev database. But it turns out it can't be uninstalled. It hangs the instance and never commits the changes.

So, basically, there are two bugs with this module:

- It breaks native openerp tests routine.
- It can't be uninstalled.
